include/App/Services.hpp

```
@@ -0,0 +1,75 @@
+    [[nodiscard]] T& GetService() const
+    {
+        auto it = m_Services.find(typeid(T));
+
+        if (it == m_Services.end())
+            throw std::runtime_error("Service not registered!");
+
+        return *static_cast<T*>(it->second.get());
+    }
+
+    template <typename T>
+    [[nodiscard]] bool ServiceExists() const
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)